packages/client/src/generation/utils.ts

```diff
@@ -248,28 +248,31 @@ export function getReturnType({
     const promiseClose = renderPromise ? '>' : ''
 
     return `${promiseOpen}$Types.GetResult<${name}Payload<ExtArgs>, T, '${actionName}'>${
-      isChaining ? '| Null' : ''
+      isChaining ? ' | Null' : ''
     }${promiseClose}`
   }
 
+  if (isChaining && actionName === 'findUniqueOrThrow') {
+    return `Prisma__${name}Client<${getType(
+      `$Types.GetResult<${name}Payload<ExtArgs>, T, '${actionName}'>`,
+      isList,
+    )} | ${isNullable ? 'null' : 'Null'}, ${isNullable ? 'null' : 'Null'}, ExtArgs>`
+  }
+
```

Comment:
This is the code from below, moved up and handles `findUniqueOrThrow` operations when chaining. These happen when we generate types for chainable members. It also adds `null` for optional relations, and forwards any previous nullability on from the chain via `Null`. (see other [comment](https://github.com/prisma/prisma/pull/20197/files#r1261636730)).